QWeb/internal/browser/chrome.py

```diff
@@ -60,79 +60,102 @@ def open_browser(
-            emulate_device = util.get_emulation_pref(emulation)
-            options.add_experimental_option("mobileEmulation", emulate_device)
-
-        service = Service(chromedriver_path) if chromedriver_path else Service()
-        driver = Chrome(service=service, options=options)
-
-        browser_reuse_enabled = (
-            util.par2bool(util.get_rfw_variable_value("${BROWSER_REUSE_ENABLED}")) or False
-        )
-        if browser_reuse_enabled:
-            # Write WebDriver session info to RF arguments file for re-use
-            dbg_address = driver.capabilities["goog:chromeOptions"]["debuggerAddress"]
-            write_browser_session_argsfile(dbg_address, driver.command_executor._url)  # type: ignore # pylint: disable=protected-access,line-too-long
+        return create_reused_browser(debugger_address)
 
-            # Clear possible existing global values
-            BuiltIn().set_global_variable("${BROWSER_EXECUTOR_URL}", None)
-            BuiltIn().set_global_variable("${BROWSER_DEBUGGER_ADDRESS}", None)
+    driver = create_new_browser(chromedriver_path, options, **kwargs)
+    cache_browser_session(driver)
 
     browser.cache_browser(driver)
     return driver
+
+
+def create_chrome_options(chrome_args: Optional[list[str]], **kwargs: Any) -> Options:
+    """Create Chrome options based on arguments and keyword arguments."""
+    options = Options()
+    options.add_experimental_option("excludeSwitches", ["enable-logging"])
+
+    if chrome_args:
+        if any("headless" in arg.lower() for arg in chrome_args):
+            CONFIG.set_value("Headless", True)
+        for arg in chrome_args:
+            options.add_argument(arg.lstrip())
+
+    options.add_argument("start-maximized")
+    options.add_argument("--disable-notifications")
+    options.add_argument("--disable-search-engine-choice-screen")
+
+    if "headless" in kwargs:
+        CONFIG.set_value("Headless", True)
+        options.add_argument("headless")
+
+    if "prefs" in kwargs:
+        prefs = util.parse_prefs(kwargs.get("prefs"))
+        options.add_experimental_option("prefs", prefs)
+
+    if "emulation" in kwargs:
+        emulate_device = util.get_emulation_pref(kwargs["emulation"])
+        options.add_experimental_option("mobileEmulation", emulate_device)
+
+    chrome_path = kwargs.get("chrome_path", None) or util.get_rfw_variable_value("${CHROME_PATH}")
+    if chrome_path:
+        options.binary_location = chrome_path
+
+    chrome_version = (
+        kwargs.get("browser_version", None)
+        or util.get_rfw_variable_value("${BROWSER_VERSION}")
+    )
+    if chrome_version:
+        options.browser_version = chrome_version
+
+    if user.is_root():
+        options.add_argument("no-sandbox")
+
+    return options
+
+
+def resolve_chromedriver_path(executable_path: str) -> str:
+    """Resolve the path to the ChromeDriver executable."""
+    return util.get_rfw_variable_value("${CHROMEDRIVER_PATH}") or executable_path
```

Comment:
tbh no idea of all these variables, old code. Apparently you could give chromedriver path as variable, but afaik with selenium manager this probably is not needed unless you specifically want to use some other chromium based browser or for example portable version of the browser. That part is documented, just not with the rfw variable:

# Use Chromium instead of Chrome:
OpenBrowser    http://google.com     chrome    chrome_path=/path/to/chromium/chrome.exe
OpenBrowser    http://google.com     chrome    executable_path=/path/to/my/chromedriver.exe

I will add the debug log part, but I think we should sort out all of these variables and document them,, but let's handle those in another branch/PR.

Reply:
actually even logging this is not that important, as in most cases it will be empty (default), meaning that chromedriver is either on PATH or handled by selenium manager. On the other hand, if custom path is used, it will be either in this variable or as open browser argument -> not much additional value of logging it.

@turunenm If you really want this logging to be added, I will log it but only if custom value is used.

Reply:
added logging if custom chromedriver path is given in new commit

Reply:
Please add the logging because if someone 'accidentally' declares one of the overriding rf variables in their test script it'll be really difficult to figure out why qweb is behaving unexpectedly.